bin/node-pg-migrate.ts

```diff
@@ -21,10 +21,7 @@ process.on('uncaughtException', (err) => {
   process.exit(1);
 });
 
-const crossRequire = createRequire(
-  // @ts-expect-error: ignore until esm only
-  import.meta.url || __dirname
-);
+const crossRequire = createRequire(resolve('noop.js'));
```

Comment:
**question (non-blocking):** what is this `noop.js`? I don't know this file 🤔

Reply:
no op = no operation

It is a file that does not exist and will be ignored. Only the path matters.

The same hack is used internally in `module.createRequire`:

https://github.com/nodejs/node/blob/2eff28fb7a93d3f672f80b582f664a7c701569fb/lib/internal/modules/cjs/loader.js#L1583-L1585

```js
 const proxyPath = trailingSlash ? 
   path.join(filename, 'noop.js') : 
   filename; 
```

Reply:
Interesting 🤔  
I first thought you wanted to reference somehow https://github.com/salsita/node-pg-migrate/blob/main/test/migrations/001_noop.js, but it did not make any sense to me.

1. I need to check how this behaves. Why does it not error when a file is not found/invalid. And what is the fallback behavior instead.
2. If we are really using this, we should extract it into a `const NON_EXISTING_FILE = 'noop.js'` or similar, and potentially even add JSDoc to that constant.

Reply:
I made a small test to see how `noop.js` behaves 👀 

<img width="1687" alt="image" src="https://github.com/salsita/node-pg-migrate/assets/7195563/040be881-2c22-435c-bec2-09b8cc0f4be3">

```node
❯ node index.js
metaUrl: file:///Users/cquadflieg/WPSM/wui/test-project/index.js
noopUrl: /Users/cquadflieg/WPSM/wui/test-project/noop.js
❯ 
❯ node src/index.js
metaUrl: file:///Users/cquadflieg/WPSM/wui/test-project/src/index.js
noopUrl: /Users/cquadflieg/WPSM/wui/test-project/noop.js
❯ 
❯ cd src
❯ node index.js
metaUrl: file:///Users/cquadflieg/WPSM/wui/test-project/src/index.js
noopUrl: /Users/cquadflieg/WPSM/wui/test-project/src/noop.js
❯ 
❯ node ../index.js
metaUrl: file:///Users/cquadflieg/WPSM/wui/test-project/index.js
noopUrl: /Users/cquadflieg/WPSM/wui/test-project/src/noop.js
```

so..., `import.meta.url` is always pointing to the file where the statement is written in.
But `noop` is always the path where the execution was made from. (is it similar to `process.cwd()`?)

Reply:
The file name does not need to be `noop.js`. We can use any name as it will be ignored by `Module.createRequire`. In the end, only the directory matters.

`process.cwd()` will not work, as `dirname(process.cwd())` will resolve to a level higher than desired.

 It could be ``` `${process.cwd()}/` ``` because internally `Module.createRequire` will join with `noop.js`:

```js
 const trailingSlash =
 StringPrototypeEndsWith(filename, '/') ||
 (isWindows && StringPrototypeEndsWith(filename, '\\'));

 const proxyPath = trailingSlash ?
 path.join(filename, 'noop.js') :
 filename;
```

And it should work fine on Windows too.

But I prefer something more flexible, like `resolve(process.cwd(), 'anything.js')`, in which case, due to the internal workings of `resolve`, we can omit `process.cwd()`. That's why I'm proposing something like `resolve('NON_EXISTING_FILE_OR_FOLDER')` or `resolve('_')`, but it can be whatever you prefer, as the result will be the same:

- `createRequire(resolve('NON_EXISTING_FILE_OR_FOLDER'))`
- ```createRequire(`${process.cwd()}/`) ```
- `createRequire(resolve('_'))`
- ...

Reply:
looks like even that does not work

![image](https://github.com/salsita/node-pg-migrate/assets/7195563/26cd311b-0556-469b-84a5-1bbf33854ed1)

Reply:
I tried several other things locally

Could you try again on your own to bring the `test\runner.spec.ts` green, without touching it?

Reply:
See #1208